qiskit/transpiler/layout.py

```diff
@@ -123,11 +123,11 @@ def _set_type_checked_item(self, virtual, physical):
 
     def __delitem__(self, key):
         if isinstance(key, int):
-            del self._p2v[key]
             del self._v2p[self._p2v[key]]
+            del self._p2v[key]
         elif isinstance(key, Qubit):
-            del self._v2p[key]
             del self._p2v[self._v2p[key]]
+            del self._v2p[key]
```

Comment:
This file should probably not be touched

Reply:
Why? This is clearly a bug, @hhorii found. Should we separate the PR?